lib/Runtime/PlatformAgnostic/Platform/Common/Intl.cpp

```diff
@@ -327,36 +342,39 @@ namespace Intl
         );
     }
 
-    HRESULT SetNumberFormatSignificantDigits(IPlatformAgnosticResource *resource, const uint16 minSigDigits, const uint16 maxSigDigits)
+    void SetNumberFormatSignificantDigits(IPlatformAgnosticResource *resource, const uint16 minSigDigits, const uint16 maxSigDigits)
     {
         // We know what actual type we stored in the IPlatformAgnosticResource*, so cast to it.
         auto *numFormatResource = reinterpret_cast<PlatformAgnosticIntlObject<icu::NumberFormat> *>(resource);
         icu::NumberFormat *nf = numFormatResource->GetInstance();
-        // REVIEW (doilij): Without RTTI support, we can't do dynamic_cast,
-        // which is the preferred way to know whether it was indeed created as a DecimalFormat,
-        // which is not something the API will indicate to us directly (create returns a NumberFormat which may or may not be a DecimalFormat)
-        icu::DecimalFormat *df = reinterpret_cast<icu::DecimalFormat *>(nf);
-        df->setMinimumSignificantDigits(minSigDigits);
-        df->setMaximumSignificantDigits(maxSigDigits);
-        return S_OK;
+
+        // TODO(jahorto): Determine if we could ever have a NumberFormat that isn't a DecimalFormat (and if so, what to do here)
+        icu::DecimalFormat *df = dynamic_cast<icu::DecimalFormat *>(nf);
+        if (df)
+        {
+            df->setMinimumSignificantDigits(minSigDigits);
+            df->setMaximumSignificantDigits(maxSigDigits);
+        }
+        else
+        {
+            AssertMsg(false, "Could not cast an icu::NumberFormat to an icu::DecimalFormat");
+        }
     }
 
-    HRESULT SetNumberFormatIntFracDigits(IPlatformAgnosticResource *resource, const uint16 minFracDigits, const uint16 maxFracDigits, const uint16 minIntDigits)
+    void SetNumberFormatIntFracDigits(IPlatformAgnosticResource *resource, const uint16 minFracDigits, const uint16 maxFracDigits, const uint16 minIntDigits)
     {
         // We know what actual type we stored in the IPlatformAgnosticResource*, so cast to it.
         auto *numFormatResource = reinterpret_cast<PlatformAgnosticIntlObject<icu::NumberFormat> *>(resource);
         icu::NumberFormat *nf = numFormatResource->GetInstance();
         nf->setMinimumIntegerDigits(minIntDigits);
         nf->setMinimumFractionDigits(minFracDigits);
         nf->setMaximumFractionDigits(maxFracDigits);
-        return S_OK;
     }
 
-    void SetGroupingUsed(IPlatformAgnosticResource *resource, const bool isGroupingUsed)
+    void SetNumberFormatGroupingUsed(IPlatformAgnosticResource *resource, const bool isGroupingUsed)
     {
         icu::NumberFormat *nf = reinterpret_cast<PlatformAgnosticIntlObject<icu::NumberFormat> *>(resource)->GetInstance();
```

Comment:
Might as well use the standard pattern (used above) here as well.

```
         // We know what actual type we stored in the IPlatformAgnosticResource*, so cast to it.
          auto *numFormatResource = reinterpret_cast<PlatformAgnosticIntlObject<icu::NumberFormat> *>(resource);
          icu::NumberFormat *nf = numFormatResource->GetInstance();
```

Or better, refactor to a method to call that takes `IPlatformAgnosticResource *resource` and returns `icu::NumberFormat *`